### PR TITLE
TCL: Code relocation in HttpCompact and HttpBodyFactory for TCL removal

### PR DESCRIPTION
**Pull request by chitianhao**
This is a preparation step for the removal of `RawHashTable` inside these places. I split these up or it will be hard to review. There is a weird dependency that should be solved. 

`HttpBodySetRawData` in `HttpCompact` has a hash table whose value is the type `HttpBodyTemplate` defined in `HttpBodyFactory`. `HttpBodyFactory` has a map whose value is the type `HttpBodySetRawData` defined in `HttpCompact`. 
Other than that, there is a method inside `HttpCompact` having a map as an argument whose value is type `HttpBodySetRawData` that is cast to `HttpBodySet` in `HttpBodyFactory`.

Previously, TCLhashtable has `void*` as the value type but now we need to worry about the types. So, `HttpBodySetRawData` (who is a parent class of `HttpBodySet` in `HttpBodyFactory`) is moved from `HttpCompact` to `HttpBodyFactory`. The method `determine_set_by_language` is also moved there. `test_error_page_selection` is converted to catch and move to the unit test of http.

### CODE REVIEW

**Comment by zwoop** [comment]
So, we're dropping all support for languages other than "default" ?

**Comment by SolidWallOfCode** [comment]
Looks like that code was just moved. I wonder where the testing logic went, though.

**Comment by chitianhao** [comment]
The exact same thing is just moved.

And the test should be removed because the interface is totally gone and the original test is mostly testing things that are not being used now. The `determine_set_by_language` is a private method and it doesn't make sense to make it public only to test it.

**Comment by chitianhao** [comment]
Any other suggestions? @SolidWallOfCode @zwoop

**Comment by chitianhao** [comment]
So the test is converted to catch added back to unit_test of http.

**Comment by randall** [comment]
[approve ci]

File: proxy/http/Makefile.am [hunk 1]
```diff
@@ -98,6 +99,26 @@ test_ForwardedConfig_LDADD = \
 	$(top_builddir)/src/tscore/libtscore.la \
 	$(top_builddir)/src/tscpp/util/libtscpputil.la
 
+test_error_page_selection_CPPFLAGS = $(AM_CPPFLAGS)\
```
Comment (SolidWallOfCode) [inline proxy/http/Makefile.am]: This should be more generic, e.g. something like "test_proxy_http". Generally we group unit tests per directory, as a compromise between completely separate (which would be a lot of compilation overhead) and all unit tests in one executable.

File: proxy/http/HttpBodyFactory.h [hunk 2]
```diff
@@ -100,6 +100,22 @@ class HttpBodyTemplate
...
 
+////////////////////////////////////////////////////////////////////////
+//
+//      class HttpBodySetRawData
+//
+//      Raw data members of HttpBodySet
+//
+////////////////////////////////////////////////////////////////////////
+
+struct HttpBodySetRawData {
```
Comment (SolidWallOfCode) [inline proxy/http/HttpBodyFactory.h]: As long as this is being updated, add in line initializers. If there's a "bad" magic value, initialize `magic` to that.

File: proxy/http/Makefile.am (hunk 1 above)
Comment (chitianhao) [reply proxy/http/Makefile.am]: OK, I'll add a main unit test file under this directory and group them up.

File: proxy/http/HttpBodyFactory.h (hunk 2 above)
Comment (chitianhao) [reply proxy/http/HttpBodyFactory.h]: Yeah, I'll add `unsigned int magic = 0` to it.

File: proxy/http/Makefile.am (hunk 1 above)
Comment (chitianhao) [reply proxy/http/Makefile.am]: Looks like I got some duplicate symbols from test_ForwardedConfig_mocks when I put them together.
```
duplicate symbol _ioBlockAllocator in:
    unit_tests/test_proxy_http-test_ForwardedConfig_mocks.o
    ../../iocore/eventsystem/libinkevent.a(IOBuffer.o)
duplicate symbol _ioBufAllocator in:
    unit_tests/test_proxy_http-test_ForwardedConfig_mocks.o
    ../../iocore/eventsystem/libinkevent.a(IOBuffer.o)
duplicate symbol _ioDataAllocator in:
    unit_tests/test_proxy_http-test_ForwardedConfig_mocks.o
    ../../iocore/eventsystem/libinkevent.a(IOBuffer.o)
duplicate symbol __ZN13IOBufferBlock4freeEv in:
    unit_tests/test_proxy_http-test_ForwardedConfig_mocks.o
    ../../iocore/eventsystem/libinkevent.a(Inline.o)
duplicate symbol __ZN12IOBufferData4freeEv in:
    unit_tests/test_proxy_http-test_ForwardedConfig_mocks.o
    ../../iocore/eventsystem/libinkevent.a(Inline.o)
duplicate symbol __ZN13IOBufferBlockC2Ev in:
    unit_tests/test_proxy_http-test_ForwardedConfig_mocks.o
    ../../iocore/eventsystem/libinkevent.a(Inline.o)
duplicate symbol __ZN13IOBufferBlockC1Ev in:
    unit_tests/test_proxy_http-test_ForwardedConfig_mocks.o
    ../../iocore/eventsystem/libinkevent.a(Inline.o)
duplicate symbol __ZTV13IOBufferBlock in:
    unit_tests/test_proxy_http-test_ForwardedConfig_mocks.o
    ../../iocore/eventsystem/libinkevent.a(Inline.o)
duplicate symbol __ZTS13IOBufferBlock in:
    unit_tests/test_proxy_http-test_ForwardedConfig_mocks.o
    ../../iocore/eventsystem/libinkevent.a(Inline.o)
duplicate symbol __ZTI13IOBufferBlock in:
    unit_tests/test_proxy_http-test_ForwardedConfig_mocks.o
    ../../iocore/eventsystem/libinkevent.a(Inline.o)
duplicate symbol __ZTV12IOBufferData in:
    unit_tests/test_proxy_http-test_ForwardedConfig_mocks.o
    ../../iocore/eventsystem/libinkevent.a(Inline.o)
duplicate symbol __ZTS12IOBufferData in:
    unit_tests/test_proxy_http-test_ForwardedConfig_mocks.o
    ../../iocore/eventsystem/libinkevent.a(Inline.o)
duplicate symbol __ZTI12IOBufferData in:
    unit_tests/test_proxy_http-test_ForwardedConfig_mocks.o
    ../../iocore/eventsystem/libinkevent.a(Inline.o)
duplicate symbol _thread_freelist_low_watermark in:
    unit_tests/test_proxy_http-test_ForwardedConfig_mocks.o
    ../../iocore/eventsystem/libinkevent.a(ProxyAllocator.o)
duplicate symbol _thread_freelist_high_watermark in:
    unit_tests/test_proxy_http-test_ForwardedConfig_mocks.o
    ../../iocore/eventsystem/libinkevent.a(ProxyAllocator.o)
duplicate symbol __ZN6Thread15thread_data_keyE in:
    unit_tests/test_proxy_http-test_ForwardedConfig_mocks.o
    ../../iocore/eventsystem/libinkevent.a(Thread.o)
ld: 16 duplicate symbols for architecture x86_64
```

**Comment by chitianhao** [comment]
@SolidWallOfCode Updated. And `test_ForwardedConfig_mocks.cc` which serves no use is removed.

**Pull request merged**